src/coreclr/jit/assertionprop.cpp

```diff
@@ -857,12 +857,27 @@ void Compiler::optPrintAssertion(AssertionDsc* curAssertion, AssertionIndex asse
                 if (curAssertion->op1.kind == O1K_EXACT_TYPE)
                 {
                     ssize_t iconVal = curAssertion->op2.u1.iconVal;
-                    printf("Exact Type MT(0x%p %s)", dspPtr(iconVal), eeGetClassName((CORINFO_CLASS_HANDLE)iconVal));
+                    if (IsTargetAbi(CORINFO_NATIVEAOT_ABI) || opts.IsReadyToRun())
```

Comment:
nit: `IsReadyToRun` is true for NAOT too, so `IsTargetAbi(CORINFO_NATIVEAOT_ABI)` is redundant. I presume a better way is to rely on `compiler->opts.compReloc`.